### ISSUE TITLE
Incorrect behavior in `resample` when raster has a window set

### ISSUE THREAD

**Issue by dfriend21** [rspatial/terra#1433]
I've come across some incorrect behavior in `resample()` that seems to be occurring when the raster being resampled has a window set. Running `resample()` the first time works and produces the expected result. However, rerunning the code produces a raster without values. Here's a reproducible example: 

``` r
library(terra)
#> terra 1.7.73
Sys.info()[1:2]
#>   sysname   release 
#> "Windows"  "10 x64"

ext <- ext(6, 6.2, 49.6, 49.8)
elev <- rast(system.file("ex/elev.tif", package = "terra"))
elev_sm <- rast(sources(elev), win = ext)

# works
rs <- resample(elev_sm, elev)
hasValues(rs)
#> [1] TRUE

# doesn't work
rs <- resample(elev_sm, elev)
hasValues(rs)
#> [1] FALSE
``` 
<sup>Created on 2024-02-19 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by kadyb** [comment]
BTW: This seems to work if `elev_sm` is in memory.

```r
ext <- ext(6, 6.2, 49.6, 49.8)
elev <- rast(system.file("ex/elev.tif", package = "terra"))
elev_sm <- rast(sources(elev), win = ext)
set.values(elev_sm)

rs <- resample(elev_sm, elev)
hasValues(rs)
#> [1] TRUE

rs <- resample(elev_sm, elev)
hasValues(rs)
#> [1] TRUE
```

**Comment by sandra-ab** [comment]
I have noticed the same issue within a loop. 

My "fix" is to remove terra temporary files after each iteration:
`tmpFiles(remove=TRUE)`

**Comment by owenssam1** [comment]
I am having exactly the same issue where I did not have it before. When I reach the resample step in a loop of edits to a raster, it erases the data values on the raster. If you use `terra::hasValues() ` right after the resample step, it is FALSE. I think this is a bug in a newer version of terra because I have not run this code for about a year now. I have no idea how to fix this. I tried the suggestion of @sandra-ab, but it seemed to erase some essential data for terra to work in my session.
